### ISSUE TITLE
Suggestion: Show examples of the components in the demo just like material design does

### ISSUE THREAD

**Issue by EduVencovsky** [SamProf/MatBlazor#568]
Hello guys, it's been a while I haven't done anything to contribute to this repo, but I came with an idea.

Sometimes when you are in the docs, you see a huge page with a lots of examples and sometimes there is some combinations of parameters that aren't shown in the demo. One example is the `TextField` where there is a lot of parameters, but the demo don't show all of it.

So to solve this, I thought of doing something like the [material design web components demo](https://material.io/components/text-fields), where you have checkboxes, tabs and radio buttons for changing the parameters and enabling you to see all parameters combine.

![image](https://user-images.githubusercontent.com/37600269/82834644-1dc6b200-9e98-11ea-9945-860aa033fe0c.png)
 
![image](https://user-images.githubusercontent.com/37600269/82834692-3931bd00-9e98-11ea-8894-44e71a4af841.png)

This would be something very useful for the `TextField` if you only want to se the design and if you want to see the code, you can go to de other examples.

It will also make the documentation look much better.

In short, my suggestion is:

- Create a container component just like the [material design web components demo](https://material.io/components/text-fields) that will hold the component, radio buttons, checkboxes and have tabs
- Initially use this on the `TextField` component (because it's probably the one with most Parameters)
- Apply this to other components that have a lot of parameters or apply in the same components that have these options in the material demo.

**Comment by lindespang** [comment]
Hi! I think this is a great idea. Maybe you can put up a draft and see how it looks? You can branch from my branch `lindespang/matblazor_redesign` and take it from there (It has not been merged yet but me and Samprof has agreed on the design)

**Comment by henon** [comment]
Awesome idea @EduVencovsky. Exactly what I was suggesting today in #603. Here is what I wrote:

> I have a suggestion: when you are a new user checking out the lib you want to see how the components look and behave, so you click through all the components. But the first thing you see is always the table of properties and every time you have to scroll down to the examples. So I think it would be a good thing to add a small but representative example to the top of each component page. Here is an example how Vuetify does it:

![image](https://user-images.githubusercontent.com/44090/84875632-84e61980-b086-11ea-8ee4-7107be6a5821.png)

or MDC:

![image](https://user-images.githubusercontent.com/44090/84875720-a21ae800-b086-11ea-8fbd-b37426fb78b8.png)

or Skclusive Material Components, another good looking MDC library for Blazor:

![image](https://user-images.githubusercontent.com/44090/84875822-c5de2e00-b086-11ea-9ce8-12b04d25fbfc.png)

**Comment by EduVencovsky** [comment]
@lindespang great! I will work on that. Today I'm kind of busy but this week I will work something out!
